src/BuildScriptGeneratorCli/Commands/DetectCommand.cs

```diff
@@ -0,0 +1,123 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+using Microsoft.Oryx.Detector;
+using Newtonsoft.Json;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Detect all platforms and versions in the given app source directory.")]
+    internal class DetectCommand : CommandBase
+    {
+        public const string Name = "detect";
+
+        [Argument(0, Description = "The source directory. If no value is provided, the current directory is used.")]
+        [DirectoryExists]
+        public string SourceDir { get; set; }
+
+        [Option("--json", Description = "Output the detected platform data in JSON format.")]
```

Comment:
what is the format if this is not provided? We should probably use --output=json instead, so we can add more formats later and prevent conflicts like both `--json` and say `--table` (or any other format) be passed at the same time

Reply:
If the format is not provided, then by default it'll simply print the detection results as a list to console. Users could choose only one of the format, so it won't be cases that we having both formats passed in at the same time? yeah agree, we could do this --output=json instead of having them separately, will change it